src/hardware/sblaster.cpp

```diff
@@ -2011,32 +2094,41 @@ class SBLASTER final : public Module_base {
 		sb.mixer.enabled=section->Get_bool("sbmixer");
 		sb.mixer.stereo=false;
 
-		Find_Type_And_Opl(section,sb.type,oplmode);
-		configure_filters(section, oplmode);
+		sb.type = find_sbtype();
+		oplmode = find_oplmode();
 
 		switch (oplmode) {
 		case OplMode::None:
 			WriteHandler[0].Install(0x388,
 			                        adlib_gusforward,
 			                        io_width_t::byte);
 			break;
+
 		case OplMode::Cms:
 			WriteHandler[0].Install(0x388,
 			                        adlib_gusforward,
 			                        io_width_t::byte);
 			CMS_Init(section);
 			break;
+
 		case OplMode::Opl2:
 			CMS_Init(section);
 			[[fallthrough]];
+
 		case OplMode::DualOpl2:
 		case OplMode::Opl3:
 		case OplMode::Opl3Gold:
 			OPL_Init(section, oplmode);
-			configure_opl_filter();
+
+			auto opl_channel = MIXER_FindChannel("OPL");
+			const std::string opl_filter_prefs = section->Get_string(
+			        "opl_filter");
+			configure_opl_filter(opl_channel, opl_filter_prefs, sb.type);
```

Comment:
I'm not sure how this `case:` is tolerating creating new variables without being scoped inside `{ ... }`?

Reply:
Maybe because it's the last case block that introduces variables? I've added an explicit scope anyway, it's better that way.